TFT/src/User/API/Printing.c

```diff
@@ -240,11 +240,26 @@ void endPrinting(void)
       break;
   }
   infoPrinting.printing = infoPrinting.pause = false;
+  infoMenu.cur = 0;
+  request_M27(0);
   powerFailedClose();
   powerFailedDelete();
   if(infoSettings.send_end_gcode == 1){
     sendPrintCodes(1);
   }
+  char tempstr1[140];
+  char tempstr2[70];
+  u8  hour = infoPrinting.time/3600,
+      min = infoPrinting.time%3600/60,
+      sec = infoPrinting.time%60;
+  float meter = ((infoFile.source == BOARD_SD) ? coordinateGetAxisActual(E_AXIS) : coordinateGetAxisTarget(E_AXIS))/1000;
```

Comment:
This gives you the real filament length used only if absolute positioning is used and only if it's a small print because many slicers reset the E after a few meters to prevent floating point rounding errors in the firmware.